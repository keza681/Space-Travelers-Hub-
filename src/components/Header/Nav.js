import { NavLink } from 'react-router-dom';
import logoo from '../../assets/logo.png';

const Nav = () => (
  <div className="nav">
    <nav className="myNav">
      <h1 className="header">
        <img src={logoo} className="myLogo" alt="logo" width="70px" height="70px" />
        Space Traveller&apos;s Hub
      </h1>
      <ul className="nav-link">
        <NavLink activeclassname="selected" to="/">Rockets</NavLink>
        <NavLink activeclassname="selected" id="mission" to="/missions">Missions</NavLink>
        <NavLink activeclassname="selected" to="/myprofile">My Profile</NavLink>
      </ul>
    </nav>
  </div>
);

export default Nav;
