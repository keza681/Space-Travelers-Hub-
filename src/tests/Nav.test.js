import React from 'react';
import renderer from 'react-test-renderer';
import { HashRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../Redux/configureStore';
import Nav from '../components/Header/Nav';

describe('Header containing the Navbar Component', () => {
  it('Should redenders NavBar Component succefully', () => {
    const app = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Nav />
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(app).toMatchSnapshot();
  });
});
