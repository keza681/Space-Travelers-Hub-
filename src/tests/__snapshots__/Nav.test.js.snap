// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header containing the Navbar Component Should redenders NavBar Component succefully 1`] = `
<div
  className="nav"
>
  <nav
    className="myNav"
  >
    <h1
      className="header"
    >
      <img
        alt="logo"
        className="myLogo"
        height="70px"
        src="logo.png"
        width="70px"
      />
      Space Traveller's Hub
    </h1>
    <ul
      className="nav-link"
    >
      <a
        activeclassname="selected"
        aria-current="page"
        className="active"
        href="#/"
        onClick={[Function]}
      >
        Rockets
      </a>
      <a
        activeclassname="selected"
        className=""
        href="#/missions"
        id="mission"
        onClick={[Function]}
      >
        Missions
      </a>
      <a
        activeclassname="selected"
        className=""
        href="#/myprofile"
        onClick={[Function]}
      >
        My Profile
      </a>
    </ul>
  </nav>
</div>
`;
